keys_storage.go

```diff
@@ -1,19 +1,30 @@
 package doubleratchet
 
+import (
+	"bytes"
+	"sort"
+)
+
 // KeysStorage is an interface of an abstract in-memory or persistent keys storage.
 type KeysStorage interface {
 	// Get returns a message key by the given key and message number.
 	Get(k Key, msgNum uint) (mk Key, ok bool, err error)
 
 	// Put saves the given mk under the specified key and msgNum.
-	Put(k Key, msgNum uint, mk Key) error
+	Put(sessionID []byte, k Key, msgNum uint, mk Key, keySeqNum uint) error
 
 	// DeleteMk ensures there's no message key under the specified key and msgNum.
 	DeleteMk(k Key, msgNum uint) error
 
 	// DeletePk ensures there's no message keys under the specified key.
 	DeletePk(k Key) error
 
+	// DeleteOldMessageKeys deletes old message keys for a session
+	DeleteOldMessageKeys(sessionID []byte, deleteUntil uint) error
+
+	// TruncateMessageKeys truncates the number of keys to maxKeysPerSession
+	TruncateMessageKeys(sessionID []byte, maxKeysPerSession uint) error
```

Comment:
`maxKeysPerSession` is a number of keys but it has `uint` type. In `DeleteOldMessagesKeys` we have `deleteUntil` also as `uint` but it's a bit unclear what each argument represents.

I would suggest:
```go
DeleteOldMessageKeys(sessionID []byte, deleteUntilSeqKey uint) error
DeleteOldMessageKeys(sessionID []byte, maxKeys int) error
```